chains/types.ts

```diff
@@ -1,27 +1,337 @@
-import { ArrayOneOrMore } from '../core/types/array.ts';
-
-export type ChainId =
-  | 'AlephZero'
-  | 'AlephZeroTestnet'
-  | 'Astar'
-  | 'ContractsRococo'
-  | 'Development'
-  | 'Khala'
-  | 'Phala'
-  | 'PhalaTestnet'
-  | 'Rococo'
-  | 'Shibuya'
-  | 'Shiden';
-
-export interface IChain {
+export type ChainName = string;
+
+export type ChainId = ProductionChainId | TestChainId;
+
+export type ProductionChainId =
+  | 'polkadot'
+  | 'kusama'
+  | '3-dpass'
+  | 'acala'
+  | 'ajuna'
+  | 'aleph'
+  | 'altair'
+  | 'amplitude'
+  | 'ares-odyssey'
+  | 'astar'
+  | 'automata'
+  | 'aventus'
+  | 'bajun'
+  | 'basilisk'
+  | 'bifrost-kusama'
+  | 'bifrost-polkadot'
+  | 'bitcountry-pioneer'
+  | 'bitgreen'
+  | 'bittensor'
+  | 'calamari'
+  | 'centrifuge-polkadot'
+  | 'chainx'
+  | 'clover'
+  | 'coinversation'
+  | 'competitors-club'
+  | 'composable-finance'
+  | 'crab-kusama'
+  | 'crab-standalone'
+  | 'creditcoin'
+  | 'crown-sterling'
+  | 'crust'
+  | 'crust-parachain'
+  | 'darwinia-polkadot'
+  | 'darwinia-polkadot-2'
+  | 'darwinia-standalone'
+  | 'debio'
+  | 'dock-pos-mainnet'
+  | 'dorafactory'
+  | 'edgeware'
+  | 'efinity-polkadot'
+  | 'efinity-standalone'
+  | 'encointer'
+  | 'equilibrium-polkadot'
+  | 'frequency'
+  | 'geminis'
+  | 'genshiro-kusama'
+  | 'gm'
+  | 'hanonycash'
+  | 'hashed'
+  | 'heiko-kusama'
+  | 'hydra'
+  | 'imbue'
+  | 'integritee-kusama'
+  | 'integritee-polkadot'
+  | 'interlay'
+  | 'ipci'
+  | 'joystream'
+  | 'kabocha'
+  | 'kapex'
+  | 'karura'
+  | 'khala'
+  | 'kico'
+  | 'kilt-spiritnet'
+  | 'kintsugi'
+  | 'kpron'
+  | 'kulupu'
+  | 'kusama-bridge-hub'
+  | 'kusari'
+  | 'kylin'
+  | 'listen'
+  | 'litentry'
+  | 'litmus'
+  | 'logion'
+  | 'loom-network'
+  | 'luhn'
+  | 'mangata'
+  | 'manta'
+  | 'mars'
+  | 'mathchain'
+  | 'minix'
+  | 'moonbeam'
+  | 'moonriver'
+  | 'myriad'
+  | 'neatcoin'
+  | 'nftmart'
+  | 'nodle-polkadot'
+  | 'oak'
+  | 'odyssey'
+  | 'omnibtc'
+  | 'origintrail-parachain'
+  | 'parallel'
+  | 'pendulum'
+  | 'phala'
+  | 'picasso'
+  | 'pichiu'
+  | 'polkadex-polkadot'
+  | 'polkadex-standalone'
+  | 'polkadot-collectives'
+  | 'polkasmith'
+  | 'polymesh'
+  | 'quartz'
+  | 'riochain'
+  | 'riodefi'
+  | 'robonomics-kusama'
+  | 'sakura'
+  | 'shadow-kusama'
+  | 'sherpax'
+  | 'shiden-kusama'
+  | 'snow'
+  | 'sora-ksm'
+  | 'sora-substrate'
+  | 'spanner'
+  | 'stafi'
+  | 'statemine'
+  | 'statemint'
+  | 'subdao'
+  | 'subgame-kusama'
+  | 'subgame-polkadot'
+  | 'subgame-standalone'
+  | 'subsocial-x'
+  | 'swapdex'
+  | 'tanganika'
+  | 'ternoa'
+  | 'thebifrost-mainnet'
+  | 'tinker'
+  | 'trustbase'
+  | 'turing'
+  | 'uniarts'
+  | 'unique'
+  | 'unorthodox'
+  | 'westlake'
+  | 'xx'
+  | 'zeitgeist'
+  | 'zero';
+
+export type TestChainId =
```

Comment:
This type can potentially be inferred dynamically via the using the `as const` typescript syntax to make the hardcoded network values readonly.

Here is an example I created: [TypeScript Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wOrHcZATAFWxmWHgF44BvAKDjgEs8AuOAcgHdjc8BaGYqRhsANHTjIAhgFtgrNlhz5R4yQiQBXZDHkAqAEwBWQwEYAnCvqoNAI1QJJyAKpQANvIAWMGGFTMA9P5cSngAdNZ2DsihDBD+logekgxoYMAIAIpQLu7sXj5+gbIwkniSJaFgklAMMACeMXEAjlBB3PgA+qhpCKiVyADmCa7lxACyZKXlklk5zOL0bPm+Af7FUxVVNfWN-i1tIR3rZSUdIwLooZJggwlQYL2sANoL7ByohQcC+Lz3CJUQVwAa1KEBgjQSi3en2C3zC1wYoRQADMUpJXLUGrF-GBbBiELx3pC3h9Vn9KikgcgIHhgLtYTx-KAwK5oMAoMTOKTAgyfuS8FxXBioKEkNJOdDVry+H9+NEBcAhQwRWKJdz-PywQIoH0hBxoECvjw1Z9yQwbGBQnqDUblGJ6ABdMQAXzgklQiBQ6AA3DQaMzoPAkGh4AAlCBIJBEdBCOCUWj0JjyKAR1P8QRkBJSWTycORiAJNSabR6IymCz2uAReyOOZ5bwrQIp-PhWw16LYu4PPxwJ5c02ppC-B4A4Gg8GxNhOmiu92ekO+v3B9BwaMkMge+N51NroQiRRw3dkZ00eppOAAOTI+qgQIvMmAcbgQOAdQgyLgZ+A79XGZgqF9L9f3QABhJIUgASTwJ8vx-I9-yeK8YBvO8HwdPsmCnGggA)

[Article explaining usage of `as const`](https://medium.com/lean-coders/understand-const-as-const-and-readonly-in-typescript-a-comprehensive-explanation-1b54f24b35d6)

Reply:
This is much better 👍🏼